hummingbot/hummingbot
# feat/enable anonymous metrics in all branches

**Before submitting this PR, please make sure**:


**A description of the changes proposed in the pull request**:
Refactor of the logic that collects anonymous trade volume metrics and sends the information to Hummingbot Foundation. The previous logic that associated metrics collection to processing the logs has been removed. The new logic associates the trades volume metrics collection directly to the connectors, and used the events to gather trades (or order fills).
The configuration value `heartbeat_enabled` was renamed to `anonymized_metrics_enabled`.
The configuration value `heartbeat_interval_min` was renamed to `anonymized_metrics_interval_min`.


**Tests performed by the developer**:
Added new unit tests.
Executed a Pure Market Making strategy connected to Binance spot. I validated that the metrics appeared correctly in Datadog dashboard for metric `filled_usdt_volume`


**Test summary**:
- Created a pure_mm strategy and went through initial prompts successfully
- Confirmed wallet and exchange balances are reflecting correct
- Started the strategy without issues, confirmed orders are created and cancelled based on config
- Ran a test bot for hours without errors
- Confirmed in logs that there are no errors when orders are filled
- Trade information are correct (price, size) based on history command output
- Compared trades in client and exchange matched.
- Confirm metrics are being captured and converted to USD
- Confirmed metric interval is working as expected.
- Confirm rate_oracle are working as expected.
- Repeat steps and apply different strategies and connectors that are supported, and confirmed working as well.
- Confirmed working on USDT and non-USDT assets.
- Confirmed working on partial and full filled orders.


**Tips for QA testing**:
Check that metrics collection is active (`anonymized_metrics_enabled` should be configured with value True). The collection frequency can be increased by reducing the value of `anonymized_metrics_interval_min` (set it to 0.2 for example). You will need the value of the configuration `instance_id` to filter results in Datadog dashboard.


This PR comes from https://github.com/CoinAlpha/hummingbot/pull/129
